modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/Diagram.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+
+import '../style/diagram.scss';
+import DiagramFooter from './DiagramFooter';
+import DiagramHeader from './DiagramHeader';
+import ImagePins from './ImagePins';
+
+const Diagram = ({
+	enablePanZoom,
+	enableResetZoom,
+	image,
+	imageSettings,
+	navigationController,
+	spritemap,
+	zoomController,
+}) => {
+	const [resetZoom, setResetZoom] = useState(false);
+	const [zoomInHandler, setZoomInHandler] = useState(false);
+	const [zoomOutHandler, setZoomOutHandler] = useState(false);
+	const [changedScale, setChangedScale] = useState(false);
+	const [scale, setScale] = useState(1);
+	const [selectedOption, setSelectedOption] = useState(1);
+
+	return (
+		<div className="diagram mx-auto">
+			<DiagramHeader setSelectedOption={setSelectedOption} />
```

Comment:
Taking a look on [DiagramHeader implementation ](https://github.com/liferay-frontend/liferay-portal/pull/1162/files#diff-389d64c0c9294f5190c52f7836b0c4795362ca02cdb44d5186ea2532c8265abeR15) this component don't allow receive any properties like you're doing in `setSelectedOption`.

Reply:
Sorry you're right. this was a refuse from the second part of this story